services/geocoding-v6.js

```diff
@@ -0,0 +1,239 @@
+ * @param {String} [config.address_line1] - A string including address_number and street. These values can alternatively be provided as separate parameters. (Stuctured Input specific field)
+ * @param {String} [config.address_number] - The number associated with the house (Stuctured Input specific field)
+ * @param {String} [config.street] - The name of the street in the address (Stuctured Input specific field)
+ * @param {String} [config.block] - In some countries like Japan, the block is a component in the address (Stuctured Input specific field)
+ * @param {String} [config.place] - Typically these are cities, villages, municipalities, etc. (Stuctured Input specific field)
+ * @param {String} [config.region] - Top-level sub-national administrative features, such as states in the United States or provinces in Canada or China. (Stuctured Input specific field)
+ * @param {String} [config.neighborhood] - Colloquial sub-city features often referred to in local parlance (Stuctured Input specific field)
+ * @param {String} [config.postcode] - Postal codes used in country-specific national addressing systems. (Stuctured Input specific field)
+ * @param {String} [config.locality] - Official sub-city features (Stuctured Input specific field)
+ * @param {boolean} [config.autocomplete=true] - Return autocomplete results or not.
+ * @param {String} [config.worldview="us"] - Filter results to geographic features whose characteristics are defined differently by audiences belonging to various regional, cultural, or political groups.
+ * @return {MapiRequest}
+ *
+ * @example
+ * geocodingClient.forwardGeocode({
+ *   query: 'Paris, France',
+ *   limit: 2
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const match = response.body;
+ *   });
+ *
+ * @example
+ * // geocoding in structured input mode
+ * geocodingClient.forwardGeocode({
+ *   mode: 'structured',
+ *   address_number: '12',
+ *   street: 'Main str.'
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const match = response.body;
+ *   });
+ *
+ * @example
+ * // geocoding with proximity
+ * geocodingClient.forwardGeocode({
+ *   query: 'Paris, France',
+ *   proximity: [-95.4431142, 33.6875431]
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const match = response.body;
+ *   });
+ *
+ * // geocoding with countries
+ * geocodingClient.forwardGeocode({
+ *   query: 'Paris, France',
+ *   countries: ['fr']
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const match = response.body;
+ *   });
+ *
+ * // geocoding with bounding box
+ * geocodingClient.forwardGeocode({
+ *   query: 'Paris, France',
+ *   bbox: [2.14, 48.72, 2.55, 48.96]
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const match = response.body;
+ *   });
+ */
+GeocodingV6.forwardGeocode = function(config) {
+  v.assertShape({
+    query: v.required(v.string),
```

Comment:
@VladislavMedved will `query` be required even if intending to use structured mode? Will it throw a validation error if `query` is not included?

Reply:
Good catch! I've made it required only for `standard` mode